jkube-kit/build/service/docker/src/test/java/org/eclipse/jkube/kit/build/service/docker/BuildServiceTest.java

```diff
@@ -93,4 +107,167 @@ void tagImage_whenValidImageConfigurationProvided_shouldTagImage() throws Docker
+
+      // Then
+      assertThat(mergedBuildArgs)
+          .containsEntry("VERSION", "latest")
+          .containsEntry("FULL_IMAGE", "busybox:latest")
+          .containsEntry("REPO_1", "docker.io/library")
+          .containsEntry("IMAGE-1", "openjdk");
+    }
+
+    @Test
+    void fromBuildConfigurationAndSystemPropertiesWithSameKey_shouldNotMergeBuildArgs() {
+      // Given
+      givenBuildArgsFromImageConfiguration("VERSION", "latest");
+      givenBuildArgsFromSystemProperties("docker.buildArg.VERSION", "1.0.0");
+
+      // When & Then
+      assertThatIllegalArgumentException()
+          .isThrownBy(() -> BuildService.mergeBuildArgs(imageConfiguration, mockedJKubeConfiguration))
+          .withMessage("Multiple entries with same key: VERSION=latest and VERSION=1.0.0");
+    }
+
+    @Test
+    void fromBuildConfigurationAndProjectPropertiesWithSameKey_shouldNotMergeBuildArgs() {
+      // Given
+      givenBuildArgsFromImageConfiguration("VERSION", "latest");
+      givenBuildArgsFromProjectProperties("docker.buildArg.VERSION", "1.0.0");
+
+      // When & Then
+      assertThatIllegalArgumentException()
+          .isThrownBy(() -> BuildService.mergeBuildArgs(imageConfiguration, mockedJKubeConfiguration))
+          .withMessage("Multiple entries with same key: VERSION=latest and VERSION=1.0.0");
+    }
+
+    @Test
+    void fromBuildConfigurationAndJKubeConfigurationWithSameKey_shouldNotMergeBuildArgs() {
+      // Given
+      givenBuildArgsFromImageConfiguration("VERSION", "latest");
+      givenBuildArgsFromJKubeConfiguration("VERSION", "1.0.0");
+
+      // When & Then
+      assertThatIllegalArgumentException()
+          .isThrownBy(() -> BuildService.mergeBuildArgs(imageConfiguration, mockedJKubeConfiguration))
+          .withMessage("Multiple entries with same key: VERSION=latest and VERSION=1.0.0");
+    }
+
+    private void givenBuildArgsFromImageConfiguration(String key, String value) {
+      Map<String, String> imgConfigBuildArg = new HashMap<>();
+      imgConfigBuildArg.put(key, value);
+      imageConfiguration = ImageConfiguration.builder()
+          .name("image-name")
+          .build(BuildConfiguration.builder()
+              .args(imgConfigBuildArg)
+              .build())
+          .build();
+    }
+
+    private void givenBuildArgsFromSystemProperties(String key, String value) {
+      System.setProperty(key, value);
+    }
+
+    private void givenBuildArgsFromProjectProperties(String key, String value) {
+      Properties props = new Properties();
+      props.setProperty(key, value);
+      when(mockedJKubeConfiguration.getProject().getProperties()).thenReturn(props);
+    }
+
+    private void givenBuildArgsFromJKubeConfiguration(String key, String value) {
+      Map<String, String> jkubeConfigBuildArg = new HashMap<>();
+      jkubeConfigBuildArg.put(key, value);
+      when(mockedJKubeConfiguration.getBuildArgs()).thenReturn(jkubeConfigBuildArg);
```

Comment:
nitpick, since there is only single entry in map... maybe we can use `Collections.singletonMap(key, value)` here?